crates/sui-adapter/src/execution_engine.rs

```diff
@@ -239,6 +239,11 @@ fn execute_transaction<
             // else, we're in dev-inspect mode, which lets you turn bytes into arbitrary
             // objects (including coins). this can violate conservation, but it's expected
         }
+        let conservation_result = temporary_store.check_storage_rebate_and_fees_conserved();
+        if conservation_result.is_err() {
+            // TODO: instead of panic, dump temporary store and recover...
```

Comment:
Curious for ideas on what to do here. My inclination would be:
1. Dump the writes
2. Call `temporay_storage.charge_gas` one more time

Reply:
I like the idea of dumping the writes and "retry", also an invariant violation would be good

Reply:
Dump the write and make re-entry